File: server/src/main/java/com/smartexpense/api/domain/model/Expense.java
package com.smartexpense.api.domain.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Expense {
    private UUID id;
    private UUID userId;
    private BigDecimal amount;
    private String category;
    private String photoUrl;
    private String caption;
    private LocalDateTime expenseDate;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: server/bin/main/com/smartexpense/api/domain/model/Expense.kt
package com.smartexpense.api.domain.model

import java.math.BigDecimal
import java.time.LocalDateTime
import java.util.UUID

data class Expense(
    val id: UUID? = null,
    val userId: UUID,
    val amount: BigDecimal,
    val category: String,
    val photoUrl: String,
    val caption: String? = null,
    val expenseDate: LocalDateTime,
    val createdAt: LocalDateTime? = null,
    val updatedAt: LocalDateTime? = null
)
